src/status_im/infra/market/events.cljs

```diff
@@ -0,0 +1,79 @@
+(ns status-im.infra.market.events
+  (:require
+    [status-im.infra.market.core :as core]
+    [status-im.infra.market.transform :as t]
+    [utils.number]
+    [utils.re-frame :as rf]))
+
+(def ^:const leaderboard-request-page-size 20)
+
+(rf/reg-event-fx :infra.market/start-fetching-tokens
+ (fn [{:keys [db]}]
+   (let [start-page 1]
+     {:db (assoc db :infra.market/last-fetched-page start-page)
```

Comment:
`:infra.market/last-fetched-page` is an example of data that belongs to specific layer. No other layers of app should be aware that tokens fetching is done page-by-page, because it is an implementation detail.